tests/internal/test_module.py

```diff
@@ -0,0 +1,223 @@
+from contextlib import contextmanager
+from os.path import dirname
+import sys
+
+import mock
+import pytest
+
+from ddtrace.internal.module import ModuleWatchdog
+from ddtrace.internal.module import origin
+
+
+@pytest.fixture
+def module_watchdog():
+    ModuleWatchdog.install()
+
+    yield sys.modules
+
+    ModuleWatchdog.uninstall()
+
+
+@contextmanager
+def unloaded_modules(modules):
```

Comment:
Could we instead make most (maybe even all) these tests subprocess test cases? With all this dependence/patching on the built in Python import process I'd have a lot more confidence if we restarted the interpreter from scratch each time.

Reply:
👍  makes sense